### PR TITLE
Add Key Id to Header

### PR DESCRIPTION
**Pull request by Grillz**
This PR adds Key ID to the JWT header of the ID token.

Signed-off-by: pbarker <pbarker@datapipe.com>

### CODE REVIEW

File: oauth2/handler.go [hunk 1]
```diff
@@ -250,6 +250,9 @@ func (h *Handler) AuthHandler(w http.ResponseWriter, r *http.Request, _ httprout
 		h.writeAuthorizeError(w, authorizeRequest, errors.Wrapf(fosite.ErrServerError, "Could not store session cookie: %s", err))
 		return
 	}
+	
+	// add key id to header
+	session.Headers.Add("kid", "public")
```
Comment (Grillz) [inline oauth2/handler.go]: @arekkas I know you had mentioned putting this here https://github.com/ory/hydra/blob/master/oauth2/session.go#L21 but fosite appears to not be passing the session headers though entirely. 

Even if the token headers are populated in the session, https://github.com/ory/hydra/blob/master/oauth2/handler.go#L169 appears to strip them. If we want a different solve we will likely need to move into fosite.

Comment (arekkas) [inline oauth2/handler.go]: Adding the kid [here](https://github.com/Grillz/hydra/blob/e868ef9e860003e8035bc40da5d02c6fda26fe90/oauth2/consent_strategy.go#L93) should work

**Comment by Grillz** [comment]
@arekkas I think we are good here now 👍  Thanks!

**Comment by arekkas** [comment]
Nice, looks much better now :)

**Pull request merged**